File: app/src/main/java/com/example/ep2024/presentation/ui/RegisterFragment.java
package com.example.ep2024.presentation.ui;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.fragment.NavHostFragment;

import com.example.ep2024.MainActivity;
import com.example.ep2024.R;
import com.example.ep2024.databinding.FragmentRegisterBinding;
import com.example.ep2024.domain.model.Role;
import com.example.ep2024.domain.repository.AuthRepository;
import com.example.ep2024.domain.usecase.auth.RegisterUseCase;

import java.util.List;

import javax.inject.Inject;

public class RegisterFragment extends Fragment {

    private FragmentRegisterBinding binding;

    @Inject
    AuthRepository authRepository;

    private RegisterUseCase registerUseCase;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        binding = FragmentRegisterBinding.inflate(inflater, container, false);

        // Инъекция зависимостей
        assert getActivity() != null;
        ((MainActivity) getActivity()).getAppComponent().inject(this);

        registerUseCase = new RegisterUseCase(authRepository);

        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        List<Role> roles = Role.getRolesListWithoutAdminRole();
        ArrayAdapter<Role> adapter = new ArrayAdapter<>(requireContext(), android.R.layout.simple_spinner_item, roles);
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        binding.spinnerUserRole.setAdapter(adapter);

        binding.spinnerUserRole.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                Role selectedRole = (Role) parent.getItemAtPosition(position);
                updateFormForRoleSelected(selectedRole);
            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) { }
        });

        binding.buttonRegister.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Role selectedRole = (Role) binding.spinnerUserRole.getSelectedItem();

                String email, password, confirmPassword, firstName, lastName, country,
                        city, street, zipCode, phone, companyName, description;

                switch (selectedRole)
                {
                    case USER:
                        email = binding.editTextEmail.getText().toString();
                        password = binding.editTextPassword.getText().toString();
                        confirmPassword = binding.editTextConfirmPassword.getText().toString();
                        firstName = binding.editTextFirstName.getText().toString();
                        lastName = binding.editTextLastName.getText().toString();
                        country = binding.editTextCountry.getText().toString();
                        city = binding.editTextCity.getText().toString();
                        street = binding.editTextStreet.getText().toString();
                        zipCode = binding.editTextZipCode.getText().toString();
                        phone = binding.editTextPhoneNumber.getText().toString();
                        // TODO: photos

                        break;

                    case OD:
                        email = binding.editTextEmail.getText().toString();
                        password = binding.editTextPassword.getText().toString();
                        confirmPassword = binding.editTextConfirmPassword.getText().toString();
                        firstName = binding.editTextFirstNameOD.getText().toString();
                        lastName = binding.editTextLastNameOD.getText().toString();
                        country = binding.editTextCountryOD.getText().toString();
                        city = binding.editTextCityOD.getText().toString();
                        street = binding.editTextStreetOD.getText().toString();
                        zipCode = binding.editTextZipCodeOD.getText().toString();
                        phone = binding.editTextPhoneOD.getText().toString();
                        // TODO: photos

                        break;
                    case PUP:
                        email = binding.editTextEmail.getText().toString();
                        password = binding.editTextPassword.getText().toString();
                        confirmPassword = binding.editTextConfirmPassword.getText().toString();
                        companyName = binding.editTextCompanyName.getText().toString();
                        country = binding.editTextCountryPUP.getText().toString();
                        city = binding.editTextCityPUP.getText().toString();
                        street = binding.editTextStreetPUP.getText().toString();
                        zipCode = binding.editTextZipCodePUP.getText().toString();
                        phone = binding.editTextPhonePUP.getText().toString();
                        description = binding.editTextCompanyDescription.getText().toString();
                        //TODO: photos

                        break;

                }

                if (getActivity() != null) {
                    ((MainActivity) getActivity()).updateBottomNavigationMenu();
                }

                NavHostFragment.findNavController(RegisterFragment.this)
                        .navigate(R.id.mainPageFragment);
            }
        });
    }

    private void updateFormForRoleSelected(Role role) {
        binding.layoutUser.setVisibility(View.GONE);
        binding.layoutOD.setVisibility(View.GONE);
        binding.layoutPUP.setVisibility(View.GONE);

        switch (role) {
            case OD:
                binding.layoutOD.setVisibility(View.VISIBLE);
                break;
            case PUP:
                binding.layoutPUP.setVisibility(View.VISIBLE);
                break;
            case USER:
                binding.layoutUser.setVisibility(View.VISIBLE);
                break;
        }
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: app/src/main/java/com/example/ep2024/presentation/ui/RegisterFragment.kt
package com.example.ep2024.presentation.ui

import android.R
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView
import android.widget.ArrayAdapter
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.findNavController
import com.example.ep2024.MainActivity
import com.example.ep2024.databinding.FragmentRegisterBinding

class RegisterFragment : Fragment() {

    private var _binding: FragmentRegisterBinding? = null
    private val binding get() = _binding!!

    private lateinit var userRoles: List<String>

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?): View? {
        _binding = FragmentRegisterBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        userRoles = listOf("OD", "PUP", "AK")

        val adapter = ArrayAdapter(requireContext(), R.layout.simple_spinner_item, userRoles)
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        binding.spinnerUserRole.adapter = adapter

        binding.spinnerUserRole.onItemSelectedListener = object :
            AdapterView.OnItemSelectedListener {
            override fun onItemSelected(parent: AdapterView<*>, view: View?, position: Int, id: Long) {
                val selectedRole = userRoles[position]
                updateFormForRole(selectedRole)
            }

            override fun onNothingSelected(parent: AdapterView<*>) {}
        }

        val defaultRole = binding.spinnerUserRole.selectedItem.toString()
        updateFormForRole(defaultRole)

        binding.buttonRegister.setOnClickListener {
            val selectedRole = binding.spinnerUserRole.selectedItem.toString()

            when (selectedRole) {
                "OD", "AK" -> {
                    val email = binding.editTextEmail.text.toString()
                    val password = binding.editTextPassword.text.toString()
                    val confirmPassword = binding.editTextConfirmPassword.text.toString()
                    val firstName = binding.editTextFirstName.text.toString()
                    val lastName = binding.editTextLastName.text.toString()
                    val address = binding.editTextAddress.text.toString()
                    val phone = binding.editTextPhone.text.toString()

                    //TODO: photo load

                    //TODO: reg for OD and AK
                }
                "PUP" -> {
                    val email = binding.editTextEmail.text.toString()
                    val password = binding.editTextPassword.text.toString()
                    val confirmPassword = binding.editTextConfirmPassword.text.toString()
                    val companyName = binding.editTextCompanyName.text.toString()
                    val companyAddress = binding.editTextCompanyAddress.text.toString()
                    val companyPhone = binding.editTextCompanyPhone.text.toString()
                    val companyDescription = binding.editTextCompanyDescription.text.toString()

                    //TODO: photo load

                    //TODO: reg for PUP
                }
            }

            (activity as MainActivity).isLoggedIn = true
            (activity as MainActivity).userRole = selectedRole
            (activity as MainActivity).updateBottomNavigationMenu()

            findNavController().navigate(com.example.ep2024.R.id.mainPageFragment)
        }
    }

    private fun updateFormForRole(role: String) {
        when (role) {
            "OD", "AK" -> {
                binding.layoutODandAK.visibility = View.VISIBLE
                binding.layoutPUP.visibility = View.GONE
            }
            "PUP" -> {
                binding.layoutPUP.visibility = View.VISIBLE
                binding.layoutODandAK.visibility = View.GONE
            }
            else -> {
                binding.layoutODandAK.visibility = View.GONE
                binding.layoutPUP.visibility = View.GONE
            }
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}